src/FeedRegistration.php

```diff
@@ -0,0 +1,239 @@
+
+	const ACTION_HANDLE_FEED_REGISTRATION = PINTEREST_FOR_WOOCOMMERCE_PREFIX . '-handle-feed-registration';
+
+	/**
+	 * Local Feed Configurations class.
+	 *
+	 * @var LocalFeedConfigs of local feed configurations;
+	 */
+	private $configurations;
+
+	/**
+	 * Feed File Generator Instance
+	 *
+	 * @var $feed_generator FeedGenerator
+	 */
+	private $feed_generator = null;
+
+	/**
+	 * Feed Registration.
+	 *
+	 * @since x.x.x
+	 * @param LocalFeedConfigs $local_feeds_configurations Locations configuration class.
+	 * @param FeedGenerator    $feed_generator Feed generator class.
+	 */
+	public function __construct( $local_feeds_configurations, $feed_generator ) {
+		$this->configurations = $local_feeds_configurations;
+		$this->feed_generator = $feed_generator;
+
+		add_action( self::ACTION_HANDLE_FEED_REGISTRATION, array( $this, 'handle_feed_registration' ) );
+		if ( false === as_has_scheduled_action( self::ACTION_HANDLE_FEED_REGISTRATION, array(), PINTEREST_FOR_WOOCOMMERCE_PREFIX ) ) {
+			as_schedule_recurring_action( time() + 10, 10 * MINUTE_IN_SECONDS, self::ACTION_HANDLE_FEED_REGISTRATION, array(), PINTEREST_FOR_WOOCOMMERCE_PREFIX );
+		}
+	}
+
+	/**
+	 * Check if the feed is registered based on the plugin's settings.
+	 * If not, try to register it,
+	 * Log issues.
+	 * Potentially report issues.
+	 *
+	 * Should be run on demand when settings change,
+	 * and on a scheduled basis.
+	 *
+	 * @return mixed
+	 *
+	 * @throws \Exception PHP Exception.
+	 */
+	public function handle_feed_registration() {
+
+		if ( ! $this->feed_file_exists() ) {
+			self::log( 'Feed didn\'t fully generate yet. Retrying later.', 'debug' );
+			// Feed is not generated yet, lets wait a bit longer.
+			return true;
+		}
+
+		// So far only one configuration exists. This loop is OK but the self::register_feed will need updating.
+		foreach ( $this->configurations->get_configurations() as $location => $config ) {
+
+			$feed_args = array(
+				'feed_location'             => $config['feed_url'],
+				'feed_format'               => 'XML',
+				'feed_default_currency'     => get_woocommerce_currency(),
+				'default_availability_type' => 'IN_STOCK',
+				'country'                   => $location,
+				'locale'                    => str_replace( '_', '-', determine_locale() ),
+			);
+
+			try {
+				$registered = self::register_feed( $feed_args, $location );
+
+				if ( $registered ) {
+					return true;
+				}
+
+				throw new \Exception( esc_html__( 'Could not register feed.', 'pinterest-for-woocommerce' ) );
+
+			} catch ( \Throwable $th ) {
+				self::log( $th->getMessage(), 'error' );
+				return false;
+			}
+		}
+
+	}
+
+	/**
+	 * Handles feed registration using the given arguments.
+	 * Will try to create a merchant if none exists.
+	 * Also if a different feed is registered, it will update using the URL in the
+	 * $feed_args.
+	 *
+	 * @param array $feed_args The arguments used to create the feed.
+	 * @param array $location List of served locations.
```

Comment:
I think this is probably prepping for a move to support multi countries/feeds

Reply:
@jconroy Right, `$location` should be a key in the array of feeds. See line 179.

Reply:
This is preparation for a multi-feed.

Reply:
> This is preparation for a multi-feed.

I understand, but the `$location` variable is used as the key in the `$feeds` array. With the latest changes, this is down on line 187:

https://github.com/woocommerce/pinterest-for-woocommerce/blob/310fefddb85f66bf7c4a8ddc866fd52e675554c0/src/FeedRegistration.php#L187

This is where having the `$location` variable as an array is a problem, because an array key cannot be an array. Something needs to change so that this can properly work.

Reply:
Yes, so this is more a problem with docblock for now. So far this function was called only with $location as a string identifiing a single country. I am adjusting the dockblock now, and it will be change later when more multifeed code will be added.